File: tools/benchmark/oracle-fixtures/Sample.swift
import Foundation

let maxRetries = 3

protocol Ledger {
    func post(amount: Int64)
}

struct AccountID {
    let raw: UInt64
}

enum Status {
    case active
    case frozen
}

class Account: Ledger {
    var balance: Int64

    init(balance: Int64) {
        self.balance = balance
    }

    func post(amount: Int64) {
        balance += amount
    }
}

extension Account {
    func describe() -> String {
        return "account"
    }
}

func transfer(from: Account, to: Account, amount: Int64) -> Bool {
    if amount <= 0 { return false }
    from.post(amount: -amount)
    to.post(amount: amount)
    return true
}

File: tools/benchmark/oracle-fixtures/Sample.java
package com.conducks.ledger;

import java.util.List;

public interface Ledger {
    void post(long amount);
}

class Account implements Ledger {
    private long balance;
    public static final int MAX_RETRIES = 3;

    public Account(long balance) { this.balance = balance; }

    public void post(long amount) { this.balance += amount; }

    public long getBalance() { return balance; }

    enum Status { ACTIVE, FROZEN }
}

class Transfer {
    static boolean run(Account from, Account to, long amount) {
        if (amount <= 0) return false;
        from.post(-amount);
        to.post(amount);
        return true;
    }
}
